workbench-core/environments/src/services/environmentService.ts

```diff
@@ -537,4 +541,19 @@ export class EnvironmentService {
 
     await this._aws.helpers.ddb.updateExecuteAndFormat({ key, params: { item: data } });
   }
+
+  public async listEnvironmentsForDataSet(dataSetId: string): Promise<Environment[]> {
```

Comment:
I think this should live in the [MetadataService](https://github.com/aws-solutions/solution-spark-on-aws/blob/develop/workbench-core/base/src/services/metadataService.ts). This way environments and datasets don't have to be coupled. I believe @SanketD92  has a proposal for how this information can be stored and retrieved.

Reply:
At a high level you would call `MetadataService.getRelationships('dataset', datasetId, 'environment')`. This will return a list of environments for dataset. You'll make this call in `datasetRoutes.ts`

Reply:
This is exactly what we need to do, it's just that we need to move this to the metadataService to make it more versatile.

Reply:
Done